frontend/components/page-contents/feedback-task-content/CenterFeedbackTask.content.vue

```diff
@@ -26,347 +29,250 @@ export default {
+        (
+          {
+            id: questionId,
+            name: questionName,
+            title: questionTitle,
+            required: isRequired,
+            settings: questionSettings,
+          },
+          index
+        ) => {
+          const componentType = this.factoryComponentType(
+            questionSettings.type
+          );
+
+          const formattedOptions = this.formatOptionsFromQuestionApi(
+            questionSettings.options,
+            questionName,
+            componentType
+          );
+          return {
+            id: questionId,
+            name: questionName,
+            dataset_id: this.datasetId,
+            order: index,
+            question: questionTitle,
+            options: formattedOptions,
+            is_required: isRequired,
+            component_type: componentType,
+            placeholder: questionSettings?.placeholder ?? null,
+            tooltip_message: questionSettings?.tooltip ?? null,
+          };
+        }
+      );
+    },
+    formatOptionsFromQuestionApi(options, prefixForIdOfOptions, componentType) {
+      // NOTE - the value of the options in questions from API and the value in the DatasetQuestion ORM are different
+      // - the value from the options from the questions in API could be anything (string, number, etc.)
+      // - the value from the options in the DatasetQuestion ORM is a boolean, it the state of the 'checkbox  true (if selected) or false (not selected)
+      // => this is why value is initiate as false for RATING and "" for FREE_TEXT
+
+      let defaultValueByComponent = null;
+
+      switch (componentType.toUpperCase()) {
+        case "FREE_TEXT":
+          defaultValueByComponent = false;
+          break;
+        case "RATING":
+          defaultValueByComponent = "";
+          break;
+        default:
+          console.log(`the component type ${componentType} is unknown`);
+      }
+
+      return (
+        options?.map((option, index) => {
+          const optionText = option.text ?? option.value;
+          const paramObject = {
+            index,
+            value: defaultValueByComponent,
+            text: optionText,
+            prefixForIdOfOptions,
+          };
+
+          return this.factoryOption(paramObject);
+        }) ?? [this.factoryOption({ prefixForIdOfOptions })]
+      );
+    },
+    factoryOption({
+      index = 0,
+      value = null,
+      text = "",
+      prefixForIdOfOptions,
+    }) {
+      return {
+        id: `${prefixForIdOfOptions}_${index}`,
+        value,
+        text,
+      };
+    },
+    factoryComponentType(componentType) {
+      // Here we translate the name from back to the corresponding component in front
+      let frontComponentType = null;
+
+      switch (componentType.toUpperCase()) {
+        case "TEXT":
+          frontComponentType = "FREE_TEXT";
+          break;
+        case "RATING":
+          frontComponentType = "RATING";
+          break;
+        default:
```

Comment:
If we have a 1:1 relation between backend settings and front question components, using the same naming on both sides would be more straightforward.